website/addons/mendeley/tests/test_views.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+
+from nose.tools import *  # noqa
+
+import responses
+import mock
+import unittest
+
+from tests.base import OsfTestCase
+from tests.factories import AuthUserFactory, ProjectFactory
+
+import json
+import urlparse
+
+from website.addons.mendeley.tests.factories import (
+    MendeleyAccountFactory, MendeleyUserSettingsFactory,
+    MendeleyNodeSettingsFactory
+)
+
+from website.util import api_url_for
+from website.addons.mendeley import utils
+from website.addons.mendeley import views
+from website.addons.citations.utils import serialize_folder
+
+from utils import mock_responses
+
+API_URL = 'https://api.mendeley.com'
+
+class MockNode(object):
+
+    addon = None
+
+    @property
+    def is_deleted(self):
+        return False
+
+    @property
+    def is_public(self):
+        return True
+
+    def get_addon(self, name):
+        if name == 'mendeley':
+            return self.addon
+        return None
+
+
+class MendeleyViewsTestCase(OsfTestCase):
+
+    def setUp(self):
+        super(MendeleyViewsTestCase, self).setUp()
+        self.account = MendeleyAccountFactory()
+        self.user = AuthUserFactory(external_accounts=[self.account])
+        self.account.display_name = self.user.fullname
+        self.account.save()
+        self.user_addon = MendeleyUserSettingsFactory(owner=self.user, external_account=self.account)
+        self.project = ProjectFactory(creator=self.user)
+        self.node_addon = MendeleyNodeSettingsFactory(owner=self.project)
+        self.node_addon.set_auth(external_account=self.account, user=self.user)
+        #self.user_addon.grant_oauth_access(self.node_addon, self.account, metadata={'lists': 'list'})
+        self.node = MockNode()
```

Comment:
Hm, my previous comment seems to have disappeared again.

@mfraezz Why not use `NodeFactory` here?